core/fetcher.go

```diff
@@ -0,0 +1,83 @@
+package core
+
+import (
+	"context"
+	"fmt"
+
+	rpctypes "github.com/celestiaorg/celestia-core/rpc/core/types"
+	"github.com/celestiaorg/celestia-core/types"
+	"github.com/celestiaorg/celestia-node/service/block"
+)
+
+const newBlockSubscriber = "NewBlock/Events"
+
+var newBlockEventQuery = types.QueryForEvent(types.EventNewBlock).String()
+
+type BlockFetcher struct {
+	Client
+
+	newBlockCh chan *block.Raw
+}
+
+// NewBlockFetcher returns a new `BlockFetcher`.
+func NewBlockFetcher(client Client) *BlockFetcher {
+	return &BlockFetcher{
+		Client: client,
+	}
+}
+
+// GetBlock queries Core for a `Block` at the given height.
+func (f *BlockFetcher) GetBlock(ctx context.Context, height *int64) (*block.Raw, error) {
+	raw, err := f.Block(ctx, height)
+	return raw.Block, err
+}
```

Comment:
You must check for an error here. Otherwise, the code will panic for nil pointer dereference in case the error is returned. Specifically, the raw will be nil, and dereference will happen on `raw.Block`